src/edu/csus/ecs/pc2/ui/SubmitSampleRunsPane.java

```diff
@@ -0,0 +1,877 @@
+
+		if (cdpDir != null) {
+		    updateCDPDirAndFields(cdpDir.getAbsolutePath());
+		} else {
+
+			int result = FrameUtilities.yesNoCancelDialog(this,
+					file.getAbsoluteFile() + " may not be a CDP directory, continue anyways?", "Select CDP");
+			if (result == JOptionPane.YES_OPTION) {
+	            updateCDPDirAndFields(file.getAbsolutePath());
+			}
+		}
+	}
+
+	/**
+	 * Select yaml file/entry.
+	 * 
+	 * @param dialogTitle
+	 * @return
+	 */
+	protected File selectEntry(String dialogTitle) {
+
+		JFileChooser chooser = new JFileChooser(cdpTextField.getText());
+//        chooser.setFileSelectionMode(JFileChooser.FILES_ONLY);
+		FileFilter filterYAML = new FileNameExtensionFilter("YAML document (*.yaml)", "yaml");
+		chooser.addChoosableFileFilter(filterYAML);
+
+		chooser.setAcceptAllFileFilterUsed(false);
+		chooser.setFileFilter(filterYAML);
+
+		int action = chooser.showOpenDialog(this);
+		chooser.setDialogTitle(dialogTitle);
+
+		switch (action) {
+		case JFileChooser.APPROVE_OPTION:
+			File file = chooser.getSelectedFile();
+			return file;
+		case JFileChooser.CANCEL_OPTION:
+		case JFileChooser.ERROR_OPTION:
+		default:
+			break;
+		}
+		return null;
+
+	}
+
+	void updateClientCDPPath(String path) {
+		ClientSettings settings = getContest().getClientSettings();
+		if (settings == null) {
+			settings = new ClientSettings(getContest().getClientId());
+		}
+		settings.put(CUSTOM_SUBMIT_SAMPLE_CDP_PATH, path);
+
+		getController().updateClientSettings(settings);
+	}
+
+	private void xlog(String string) {
+		System.out.println(string);
+		addLineToTextArea(string);
+	}
+
+	protected String getClientCDPPath() {
+
+		ClientSettings settings = getContest().getClientSettings();
+		if (settings != null) {
+			String path = settings.getProperty(CUSTOM_SUBMIT_SAMPLE_CDP_PATH);
+			if (path != null) {
+				return path;
+			}
+		}
+
+		return null;
+	}
+
+	/**
+	 * Reset fields back to default values
+	 * 
+	 * @param usingGui
+	 */
+	protected void resetFields(boolean isUsingGui) {
+	    
+	    
+	    selectedJudgingTypeIndexes = new int[0];
+	    selectedProblemsIndexes = new int[0];
+	    selectedLanguagesIndexes = new int[0];
+
+		checkBoxSubmitYesSamples.setSelected(true);
+		checkBoxSubmitFailingSamples.setSelected(true);
+		
+		checkBoxLanguage.setSelected(true);
+        checkBoxProblems.setSelected(true);
+        
+        submitAllJudgingTypesRadioButton.setSelected(true);
+        submitAllProblemsRadioButton.setSelected(true);
+        submitAllLanguagesRadioButton.setSelected(true);
+
+		String cdpPath = getContest().getContestInformation().getJudgeCDPBasePath();
+
+		String clientPath = getClientCDPPath();
+
+		if (clientPath != null && isUsingGui) {
+			int result = FrameUtilities.yesNoCancelDialog(this, "Overwrite locally saved CDP path with this (default) " + cdpPath,
+					"Replace CDP Path?");
+			if (result == JOptionPane.NO_OPTION) {
+				cdpPath = clientPath;
+			}
+		}
+
+		cdpTextField.setText(cdpPath);
```

Comment:
When the CDP text field is initially set, there is a message added to the text area like:
```
2 05:30:03 CDP dir is now at C:\Users\John\git\pc2v9\dougs-github-fork\samps\contests\tenprobs\config
```

It seems like when the CDP field _changes_ due to this `reset` action, a new message (giving the new CDP dir) should also be logged in the text area...

Reply:
This has become increasingly problematic as I have continued to do more testing.  See below, for example, where there is a display showing a message stating that the current CDP is one thing, and then it shows submissions from a different CDP -- no message indicating/recording that the CDP location has changed.

I _almost_ consider this an approval showstopper...